ecommerce/static/js/payment_processors/bluefin.js

```
@@ -0,0 +1,122 @@
+        form_data = new FormData($('#paymentForm')[0])
+        form_data.append("bluefin_token", res.eToken);
+
+        // Create PayConex request
+        payConexRequest(form_data, post_url);
+    })
+}
+
+//Function to POST form data to server for processing via PayConex.
+function payConexRequest(data, url){
+    $.ajax({
+        type: "POST",
+        url: url,
+        cache: false,
+        data: data,
+        processData: false,
+        contentType: false,
+    }).done(function ( response ) {
+        window.location.href = response.url;
+    }).fail(function ( data ) {
+        displayErrorMessage(gettext(
+            'An error occurred while processing your payment. Please try again.'
+        ));
+    });
+}
+
+//Function to display error-message.
+function displayErrorMessage(message) {
+    $('#messages').html(
+        `<div class="alert alert-error"><i class="icon fa fa-exclamation-triangle"></i>${message}</div>`,     
+    )
+}
+
+define([
+    'jquery',
+    'underscore.string'
+], function($, _s) {
+    'use strict';
+
+    return {
+        init: function(config) {
+            this.$paymentForm = $('#paymentForm');
+            this.$paymentForm.on('submit', function(event) {
+                submitPayment(config.postUrl);
+                return false;
+            });
+            
+            // replace card div with bluefin iframe
+            let bluefin_iframe_code = `
+                <div class="col col-sm-12">
+                    <fieldset>
+                        <div id="iframe" style="overflow:hidden;"></div>
+                    </fieldset>
+                    <div id="iframe-response"></div>
+                </div>
+            `
+            $( bluefin_iframe_code ).insertAfter( ".pci-fields" );
+            let card_div  = document.getElementsByClassName("pci-fields");
+            card_div[0].style.display = "none"
+
+            //dummy data
```

Does this dummy data need to be pushed to production? If not, then why is this hard coded?

---

Yes as there is by default full payment form (containing card details and person info) and we are using iframe instead of it's card fields so we have to set some dummy values to it otherwise it would fail in form validation.

---

But, if we push this to production, will these values show up when a user goes to the payment form?

---

style.display = "none" ( line 111 )  is hiding it. So it would not be displayed to the user, our iframe will take it's place and we will send our iframe fields after validation to the back-end.
Moreover this js file will only run in case of Bluefin processor so in case of other processors (cybersource, stripe) normal flow of Payment Form without iframe  and without any dummy values will run.

---

So shouldn't we leave these values empty or use better defaults?

---

we have to give valid dummy values to these fields. What do you mean by better defaults ?

---

I also don't get get why this dummy data is required here?